### ISSUE TITLE
Setup / installation instructions

### ISSUE THREAD

**Issue by seth-rah** [zephyrtronium/robot#76]
Hi there, trying to set up robot for a friend, but running into some issues, and I'm not sure what troubleshooting steps I should follow first as I don't see any instructions for getting robot set up for the first time

**Comment by zephyrtronium** [comment]
There is still a feature I intend to do that will eventually change setup instructions (#58), hence why there is nothing in the readme yet. That said, I'll give a rough outline of how to get started with the current version. I'm not sure what your level of expertise is, so feel free to ask follow-up questions.

1. Create an application for your instance of the bot in the [Twitch dev console](https://dev.twitch.tv/console/apps/create).
    - The OAuth Redirect URL doesn't need to exist *for now* (again pending future changes), but your config will have to match what you enter.
    - Leave the client type as *Confidential*.
2. Save the client secret Twitch shows you to a file. Keep the client ID on hand.
3. With the Go toolchain [installed and configured](https://go.dev/doc/install), run `go install github.com/zephyrtronium/robot@master`. (Without the `@master`, it will install the old version that has some performance and moderation issues.)
4. Run `robot -help` to verify it's installed.
5. Generate a secure random key. On Linux, you can do this with a command along the lines of `head -c 32 /dev/random >robot_key && chmod 0400 robot_key`, replacing `robot_key` with wherever you'd like to keep the key.
6. Modify [example.toml](https://github.com/zephyrtronium/robot/blob/master/example.toml) according to your needs.
    - Use the path to the key you generated in step 5 for the `secret` at the top of the file.
    - Use the path to the Twitch client secret you saved in step 2 for `secret` in the `[tmi]` section.
    - I highly recommend using the sqlbrain option in `[db]` rather than kvbrain.
7. Run `robot -config your.toml`. If everything is configured correctly, it should prompt you to go to a URL to authorize your Robot instance.

Once all that is done, the bot should work.

**Comment by seth-rah** [comment]
I'll give it a go later, thanks. Will let you know if I run into any hiccups

**Comment by seth-rah** [comment]
@zephyrtronium Question on this one

managed to set up the user and wanted to know, should I set up a db instance myself or is that done through the bot run?

I tried setting up the user, but not completely sure if I set everything up right myself

The user didn't auto-connect to a channel, so I made it join manually, and trying to use a say command to see if it's aware returned this error

```
time=2024-09-15T23:54:47.897+02:00 level=INFO msg="stream infos" count=0
time=2024-09-15T23:55:41.678+02:00 level=INFO msg=command level=any name=speak args=map[prompt:words] in=#finleyfisch
time=2024-09-15T23:55:41.681+02:00 level=ERROR msg="couldn't speak" err="couldn't speak: couldn't prepare term selection: sqlite: prepare \"SELECT id, suffix FROM knowledge WHERE tag = :tag AND prefix >= :lower AND prefix < :upper AND LIKELY(deleted IS NULL)\": SQL logic error: no such table: knowledge"
```

**Comment by seth-rah** [comment]
Ok, I see what I did wrong here

So now I'm at a point where it's operational but I don't seem to see it learning anything

does the channel need to be live for it to interact with the DB?

![image](https://github.com/user-attachments/assets/34686537-b468-4ce8-848f-5d2768883035)

**Comment by zephyrtronium** [comment]
Yes, the stream needs to be live for Robot to learn from it. The assumption is that offline chat is generally unmoderated, and Robot generally relies on moderators to keep its knowledge appropriate. I've considered adding a configuration option to override that behavior to help with testing an installation, but it's been low priority so far.

**Comment by seth-rah** [comment]
Gotcha, thanks for the help again

**Comment by seth-rah** [comment]
Sorry for bothering on a closed issue

Just trying to debug right now

So not sure if it's config related or not, or does robot have a minimum learning cycle

But right now robot isn't able to say anything in chat unless it's a copy-pasta. it will only repeat existing messages but never speak for itself

Any idea what might be the reason, even when stream is online

![image](https://github.com/user-attachments/assets/227d0784-27bf-4e8a-a985-39b18ac8a4bf)

**Comment by seth-rah** [comment]
The knowledge table is definitely populating, and most of the config was kept as default minus some additions

![image](https://github.com/user-attachments/assets/27b14f05-dbd6-4390-9b86-994c90a4acae)

```toml
[global.privileges]
twitch = [
  { name = 'nightbot', level = 'ignore' },
  { name = 'frostytoolbots', level = 'ignore' },
  { name = 'tangiabot', level = 'ignore' },
  { name = 'soundalerts', level = 'ignore' },
  { name = 'blerp', level = 'ignore' },
  { name = 'kofistreambot', level = 'ignore' },
  { name = 'streamelementsbot', level = 'ignore' },
]

[tmi]

cid = 'secretsshhhhh'

secret = 'D:\robot\twitchSecret'

redirect = 'http://localhost'

token = 'D:\robot\tmi_refresh'

owner = { id = '961395622', name = 'finleyfisch' }

rate = { every = 30, num = 20 }

[twitch.finleyfisch]

channels = ['#finleyfisch']

learn = 'finleyfisch'

send = 'bocchi'

block = '(?i)cucumber[^$x]'

responses = 0.02

rate = { every = 10.1, num = 2 }

copypasta = { need = 2, within = 30 }

privileges = [
  { name = 'sethrah', level = 'moderator' },
  { name = 'finleyfisch', level = 'moderator' },
]

[twitch.bocchi.emotes]
'btw make sure to stretch, hydrate, and take care of yourself <3' = 1

[twitch.bocchi.effects]
'AAAAA' = 44444
```

**Comment by seth-rah** [comment]
Oh, I think I understand what you meant with learn and send

**Comment by seth-rah** [comment]
Yeah, that seems to have fixed it >.<

We created a new user for the bot so hopefully it won't get spam blocked.

We only have it learning and living in one channel
